Sources/MuxPlayerSwift/InternalExtensions/AVPlayerItem+Mux.swift

```diff
@@ -113,3 +148,363 @@ internal extension AVPlayerItem {
+        let mvhdTypeRange = mp4Data.range(of: ShortFormMediaPlaylistGenerator.movieHeaderType)
+        guard let mvhdTypeRange, !mvhdTypeRange.isEmpty else {
+            throw ShortFormRequestError.unexpected(message: "no mvhd found in int segment")
+        }
+        
+        let boxStart = mvhdTypeRange.startIndex - 4 // 4 bytes for the size field
+        guard boxStart >= 0 else {
+            throw ShortFormRequestError.unexpected(message: "mvhd start was out of bounds")
+        }
+        let boxSize = try readInt32(data: mp4Data, at: UInt(boxStart))
+        let boxEnd = Int32(boxStart) + boxSize // not inclusive
+        guard boxEnd <= mp4Data.count else {
+            throw ShortFormRequestError.unexpected(message: "mvhd end was out of bounds")
+        }
+        
+        let timescaleStart = boxStart + ShortFormMediaPlaylistGenerator.movieHeaderTimeScaleOffset
+        let durationStart = boxStart + ShortFormMediaPlaylistGenerator.movieHeaderDurationOffset
+        let timescale = try readInt32(data: mp4Data, at: UInt(timescaleStart))
+        let duration = try readInt32(data: mp4Data, at: UInt(durationStart))
+        
+        return Double(duration) / Double(timescale)
+    }
+    
+    private func approximatelyEqual(_ lhs: Double, _ rhs: Double, tolerance: Double) -> Bool {
+        return abs(lhs - rhs) <= tolerance
+    }
+    
+    /// Reads an int32 out of the given data. The data is read big-endian because isobmff files are big-endian
+    private func readInt32(data: Data, at offset: UInt) throws -> Int32 {
+        guard data.count >= 4 && offset < data.count - 4 else {
+            throw ShortFormRequestError.unexpected(message: "readInt32: out of bounds")
+        }
+        return data[offset..<(offset + 4)].withUnsafeBytes { bytePointer in
+            return bytePointer.load(as: Int32.self).bigEndian
+        }
+    }
+    
+    /// @param originBaseURL: An aboslute URL that points to the path where segments can be found (ie, `https://shortform.mux.com/abc23/`
+    /// @param cacheProxyURL: An absolute URL that points to the path where the cache proxy is (ie, `http://127.0.0.1:1234/`)
+    init(
+        initSegment: Data,
+        originBaseURL: URL,
+        cacheProxyBaseURL: URL,
+        playlistAttributes: PlaylistAttributes
+    ) {
+        self.initSegmentData = initSegment
+        self.playlistAttributes = playlistAttributes
+        
+        // since we are not resolvingAgainstBaseURL, there's no risk of these initializers returning nil
+        self.cacheProxyBase = URLComponents(url: cacheProxyBaseURL, resolvingAgainstBaseURL: false)!
+        self.originBase = URLComponents(url: originBaseURL, resolvingAgainstBaseURL: false)!
+    }
+    
+    struct PlaylistAttributes {
+        let version: UInt
+        let targetDuration: UInt
+        let extinfSegmentDuration: Double? // assumed to be the target duration if not specified
+    }
+    
+    private class Tags {
```

Comment:
no pressure here but a few of these should be irrelevant for our use case (e.g. `#EXT-X-DISCONTINUITY`, `#EXT-X-MEDIA-SEQUENCE`, the `BYTERANGE` attribute for `#EXT-X-MAP`)

Reply:
I think `AVPlayer` needs the `#EXT-X-DISCONTINUITY` in order to play. I found this out by forgetting to add it 😓 

Good to know about the rest of the tags!